Style: implementation of College of Chemistry & College of Bioscience's undergraduate thesis style

yizeyi18 opened the pull request:
A first step to #72 .
Including changes on:
 - new control macro `\college` in `main.tex`, passed to `NKThesis` package and straightfully controls some code in `main.tex`. Implemented value including `Chem` and `Bio`.
 - new `if` in `NKThesis.sty`: `\if@NKT@Chem` and `\if@NKT@Bio`, used to control ollege specific behavior inside the package.
 - College specific behaviors: 
   - number acknowledgement in table of contents(toc) with arabic number for `Chem` or `Bio`.
   - <del>show declaration in toc for `Chem`.</del> [NKU specifies](https://chem.nankai.edu.cn/2024/0523/c24081a543996/page.htm) that declaration should not appear in toc, change here reverted.
   - remove space after acknowledgement, abstract and chapter-section-subsections in toc for `Chem` and `Bio`, remove space after toc and reference in toc for `Chem`.
   - remove line of toc in toc for `Bio`.
   - redefine numbering strategy and font size for chapter-section-subsections , remove extra spaces between numbering and title of chapter-section-subsections for `Chem` and `Bio`.
   - add new appendix command `\fulu` for `Chem`. `Chem` requires appendix title have same font size as section, so existing `\appendix` command, which is derived from `\chapter`, is incompatible here.
 - Cover alignment adjustification: moved some items to top to make head of all items aligned.

All college-specific behaviors are from templates on [NKU undergraduate thesis platform](https://nankai.co.cnki.net/) uploaded by colleges.

yizeyi18 committed:
Add swiches between style from college-specific templates; Chem and Bio implemented.

yizeyi18 committed:
style improvements and more college-specific tempelate implementation.

Tr0py commented:
Looks good!  Will review the code this weekend.

yizeyi18 committed:
No page number for decl page

Tr0py commented on main.tex:
```diff
@@ -9,7 +9,9 @@
 \usepackage{geometry}
-\usepackage[emptydoublepage]{NKThesis}   % 中文
+%分学院风格的模板。已实现的：Chem/化院，Bio/生科院
+\def\college{Bio}
```
如果不define college，会fallback到默认的计算机学院格式吗？

yizeyi18 replied:
是的，保留了计算机学院格式作为默认值。

Tr0py commented on main.tex:
```diff
@@ -112,13 +117,23 @@
-\include{acknowledgements}
-\include{resume}
-
+\ifdefstring{\college}{Chem}{
+  \include{acknowledgements}
+  \end{spacing}
+  }{
+\ifdefstring{\college}{Bio}{
+  \include{acknowledgements}
+  \end{spacing}
+  }{
+  \end{spacing}
+  \include{acknowledgements}
+  \include{resume}
+}%Bio
+}%Chem
```
我建议这样改一下，会让代码更易读：

```latex
%\ifdefstring{\college}{Chem}{\NKTdeclaration}{}
\include{abstract}
\tableofcontents
\begin{spacing}{1.5}
\include{manual}
\include{references}
\end{spacing}

% Set the spacing value based on the college
\ifthenelse{\equal{\college}{Chem} \OR \equal{\college}{Bio}}{
    \def\ackspacing{1.5}
}{
    \def\ackspacing{1}
}

\begin{spacing}{\ackspacing}
\include{acknowledgements}
\end{spacing}

% Include resume if college is neither Chem nor Bio
\ifthenelse{\equal{\college}{Chem} \OR \equal{\college}{Bio}}{}{
    \include{resume}
}
```

因为Chem和Bio的逻辑是一样的，所以最好用 if (chem or bio) 来合并逻辑。

Tr0py commented on NKThesis.sty:
```diff
@@ -781,16 +860,33 @@ Path=fonts/
-  \phantomsection
-  \addcontentsline{toc}{chapter}{\protect\numberline{摘 要}{}}
-  \markboth{摘\quad 要}{摘\quad 要} 
+  \if@NKT@Chem
+    \centerline{\zihaosi\bfseries 摘\qquad 要}
+    \phantomsection
+    \addcontentsline{toc}{chapter}{\protect 摘\qquad 要}
+    \markboth{摘\qquad 要}{摘\qquad 要}
+  \else%Chem
+    \centerline{\zihaosi\bfseries 摘\quad 要}
+    \phantomsection
+    \addcontentsline{toc}{chapter}{\protect\numberline{摘 要}{}}
+    \markboth{摘\quad 要}{摘\quad 要}
+  \fi%Chem 
```
这些代码本身很好，没有逻辑上的问题。但是代码重复会比较多，未来可能会难以维护。我能想到的一个解决办法是把这些参数提取出来，统一设定，比如
```
% if Chem zhaiyaospacing = \qquad
% if Bio zhaiyaospaceing = \quad

    \centerline{\zihaosi\bfseries 摘\zhaiyaospacing要}
    \phantomsection
    \addcontentsline{toc}{chapter}{\protect 摘\zhaiyaospacing 要}
    \markboth{摘\zhaiyaospacing 要}{摘\zhaiyaospacing 要}

yizeyi18 replied:
我赞成提出来统一设定的想法，比这个PR里的实现漂亮得多。
一个问题是，不同学院间区别可能涉及整行内容之有无，比如目录里放什么、不放什么；我认为把整个代码块拎出来做定义较好。

Tr0py replied:
> 我赞成提出来统一设定的想法，比这个PR里的实现漂亮得多。 一个问题是，不同学院间区别可能涉及整行内容之有无，比如目录里放什么、不放什么；我认为把整个代码块拎出来做定义较好。

同意，我觉得如果是“涉及整行内容之有无”的代码区别确实可以对整个代码块按照学院进行划分。

Tr0py commented:
我理解的对吗：你定义了三种College: Chem, Bio, default。根据college不同你会对章节，附录，行间距进行不同的设置。

我测试了代码，可以正常编译，也可以很方便的切换学院。

我能看到的问题是在未来如果有更多的学院需要配置，代码会变得非常冗长。因为目前使用if else 加重复定义变量实现的，所以每增加一个学员就需要增加一份code。当相同功能的code有好几份时，代码维护会变得困难。因为如果想要改变有重复的代码时需要对多处进行修改。

我觉得第一步可以是尝试将需要根据学院来定义的variable找出来，比如上面举例的`\def\ackspacing{1}`。这可以包括你修改过的 chaptername 等。我们再在某处统一定义这些变量。

有这样一套framework会方便后续的开发，是支持多学院格式很重要的一步。我们不然先把这个框架作为第一步，你觉得呢？

yizeyi18 commented:
@Tr0py 确实，用一处判断比用分散的多处判断合理得多。我可最近实现一份试试。

yizeyi18 committed:
add comments for each sections to be formatted college-by-college;
list macros that would be used for college-specific formatting.
Macros to be used are:
 - \chapNameStyle
 - \chapFormatStyle
 - \tocINtoc
 - \skipINtoc
 - \chapINtoc
 - \sectionINtoc
 - \subsectionINtoc
 - \zhixie
 - \zhaiyao
